internal/index/validation/validate.go

```diff
@@ -99,6 +99,45 @@ func ValidatePlugin(name string, p index.Plugin) error {
 	return nil
 }
 
+// ValidateReceipt checks for structural validity of the Receipt object with given
+// name.
+func ValidateReceipt(name string, r index.Receipt) error {
```

Comment:
This is pure duplication, we shouldn't do this.

I'm not sure if/why we need to validate `Receipt`s. `Plugin`s are authored by developer community; `Receipt`s aren't. Therefore it should be **undefined/unsupported behavior** trying to catch validation errors (stemmed from `Plugin`) in the `Receipt` type.

Reply:
Ya I was pretty iffy on this. I only added it because the original receipt loading called `ValidatePlugin`, I can remove it though.

Reply:
Yeah we haven't gotten a whole lot out of this.
The reason we have it on `Plugin` is because it's actually an "input" to our program. However `Receipt` is something our program writes/reads itself to store "state".